test/unit/payment_icon_test.rb

```diff
@@ -14,4 +16,40 @@ class PaymentIconTest < ActiveSupport::TestCase
     visa = PaymentIcon.where(name: 'visa').first
     assert_equal 'payment_icons/visa.svg', visa.path
   end
+
+  test 'Every payment icon record has a corresponding SVG file' do
+    PaymentIcon.all.each do |icon|
+      assert File.exist?(ICONS_DIRECTORY.join("#{icon.name}.svg"))
+    end
+  end
+
+  test 'Every payment icon SVG has a valid XML tree' do
+    PaymentIcon.all.each do |icon|
+      file = File.read(ICONS_DIRECTORY.join("#{icon.name}.svg")).freeze
+      document = Nokogiri::XML.parse(file)
+      assert document.errors.blank?, message: "The '#{icon.name}' SVG file is invalid: #{document.errors}"
```

Comment:
Shall we recommend using SVGO in the error message?

Reply:
> Does SVGO provide suggestions to fix an invalid SVG document?
Not sure actually 😊

For "\n" characters, we don't want them cause of performance right?
If so, then maybe we shouldn't allow whitespace either, which could be done with `/(^\s)|(\s$)/`

Reply:
Come to think of it, this sort of optimization might be a little overkill for our needs. I'll mention SVGO in the error message, and combined with the PR checklist containing the SVGO bit, we should be good. :+1: